src/polly/src/epoll.rs

```diff
@@ -0,0 +1,201 @@
+        const READ_HANG_UP = EPOLLRDHUP as u32;
+        /// Sets the Edge Triggered behavior for the associated file descriptor.
+        /// The default behavior is Level Triggered.
+        const EDGE_TRIGGERED = EPOLLET as u32;
+        /// Hang up happened on the associated file descriptor. Note that `epoll_wait`
+        /// will always wait for this event and it is not necessary to set it in events.
+        const HANG_UP = EPOLLHUP as u32;
+        /// There is an exceptional condition on that file descriptor. It is mostly used to
+        /// set high priority for some data.
+        const PRIORITY = EPOLLPRI as u32;
+        /// The event is considered as being "processed" from the time when it is returned
+        /// by a call to `epoll_wait` until the next call to `epoll_wait` on the same
+        /// epoll file descriptor, the closure of that file descriptor, the removal of the
+        /// event file descriptor via EPOLL_CTL_DEL, or the clearing of EPOLLWAKEUP
+        /// for the event file descriptor via EPOLL_CTL_MOD.
+        const WAKE_UP = EPOLLWAKEUP as u32;
+        /// Sets the one-shot behavior for the associated file descriptor.
+        const ONE_SHOT = EPOLLONESHOT as u32;
+        /// Sets an exclusive wake up mode for the epoll file descriptor that is being
+        /// attached to the associated file descriptor.
+        /// When a wake up event occurs and multiple epoll file descriptors are attached to
+        /// the same target file using this mode, one or more of the epoll file descriptors
+        /// will receive an event with `epoll_wait`. The default here is for all those file
+        /// descriptors to receive an event.
+        const EXCLUSIVE = EPOLLEXCLUSIVE as u32;
+    }
+}
+
+/// This is a wrapper over 'libc::epoll_event'.
+///
+/// We are using `transparent` here to be super sure that this struct and its fields
+/// have the same alignment as those from the `epoll_event` struct from C.
+#[repr(transparent)]
+#[derive(Clone, Copy)]
+pub struct Event(epoll_event);
+
+impl Deref for Event {
+    type Target = epoll_event;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl Default for Event {
+    fn default() -> Self {
+        Event(epoll_event {
+            events: 0u32,
+            u64: 0u64,
+        })
+    }
+}
+
+impl Event {
+    /// Create a new epoll_event instance with the following fields: `events`, which contains
+    /// an event mask and `data` which represents a user data variable. `data` field can be
+    /// a fd on which we want to monitor the events specified by `events`.
+    pub fn new(events: EventType, data: u64) -> Self {
+        Event(epoll_event {
+            events: events.bits(),
+            u64: data,
+        })
+    }
+
+    pub fn events(&self) -> u32 {
+        self.events
+    }
+
+    pub fn data(&self) -> u64 {
+        self.u64
+    }
+}
+
+/// Stores a file descriptor referring to an epoll instance.
```

Comment:
nit: "Wrapper over [epoll](http://man7.org/linux/man-pages/man7/epoll.7.html) functionality."

This is not just storing the file descriptor, but also offers epoll_* functionality via its methods.